### PR TITLE
[page-flip] Create @types/page-flip

### PR DESCRIPTION
**Pull request by lethyb**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by lethyb** [comment]
I asked the "page-flip" owners (@Nodlik & @btzr-io) to review and/or add any comments to this PR.

File: types/page-flip/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import { PageFlip } from "./dist/PageFlip";
+import { CanvasRender } from "./dist/Render/CanvasRender";
+import { HTMLRender } from "./dist/Render/HTMLRender";
+import { FrameAction, Orientation, Render, Shadow } from "./dist/Render/Render";
+import { FlipSetting, Settings, SizeType } from "./dist/Settings";
+import { CanvasUI } from "./dist/UI/CanvasUI";
+import { HTMLUI } from "./dist/UI/HTMLUI";
+import { SwipeData, UI } from "./dist/UI/UI";
+
+export as namespace StPageFlip;
```
Comment (peterblazejewicz) [inline types/page-flip/index.d.ts:26]:
```suggestion
export as namespace St;
```

```
// or if you're using a script tag and page-flip.browser.js:
const pageFlip = new St.PageFlip(htmlParentElement, settings);
```
https://nodlik.github.io/StPageFlip/#usage

Comment (lethyb) [reply types/page-flip/index.d.ts]: Thank you @peterblazejewicz, a following commit has been pushed

**Review by Nodlik — APPROVED** [review]

**Comment by peterblazejewicz** [comment]
@lethyb @Nodlik 
this is proper TS package, isn't it?
https://unpkg.com/browse/page-flip@2.0.6/
https://github.com/Nodlik/StPageFlip/blob/master/tsconfig.json#L4-L5
why those types on DT?

**Comment by lethyb** [comment]
Hi @peterblazejewicz,

I created those types for at least matching the latest public release of page-flip that you can see in the main page of the project, which is officially the **2.0.0**. This release doesn't specify any `declarationDir` in the _tsconfig.json_.
The `declaration` and `declarationDir` are set as compiler options in _tsconfig.json_ since the version **2.1.0**.

However, if we start with a newer version of this package, it seems the pointed declaration directory does not contain any **.ts** files. if I'm correct it should, as specified by the typescript documentation : https://www.typescriptlang.org/tsconfig#declarationDir

So in conclusion, the types seem to be missing

**Comment by peterblazejewicz** [comment]
> However, if we start with a newer version of this package, it seems the pointed declaration directory does not contain any **.ts** files (Do you get any **.ts** files ?). if I'm correct it should, as specified by the typescript documentation : [typescriptlang.org/tsconfig#declarationDir](https://www.typescriptlang.org/tsconfig#declarationDir)

Yes, I saw this (I've got `*.ts`, no `*.d.ts`). That's OK for me, I've just wanted to fact-check that problem. I can see that's OK for package owner as well.

**Review by peterblazejewicz — APPROVED** [review]
LGTM!

**Comment by lethyb** [comment]
Thank your @peterblazejewicz for reviewing this PR

**Review by orta — APPROVED** [review]
Comprehensive, nice work

**Comment by lethyb** [comment]
Ready to merge
